### PR TITLE
MRD-121_permit-risk-screen-to-proceed-when-no-summary-available:

### PR DESCRIPTION
**Pull request by bills7281**
A 404 on ARN RoSH is now handled with a log message to allow the user to proceed to the Risk screen.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RiskService.kt [hunk 1]
```diff
@@ -62,84 +62,95 @@ class RiskService(
...
+    } catch (e: WebClientResponseException.NotFound) {
+      log.info("No Risk Summary available for CRN: $crn - ${e.message}")
+      null
+    }
+  }
+
   private suspend fun handleFetchMappaApiCall(crn: String): Mappa? {
-    return try { fetchMappa(crn) } catch (e: WebClientResponseException.NotFound) {
+    return try {
+      fetchMappa(crn)
```
Comment (jonnywyatt) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RiskService.kt:76]: I'm probably reading it wrong but it looks like there's a call out to get mappa, if that 404's, the response to UI should still be a 200 with personal details?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/controller/RiskControllerTest.kt [hunk 2]
```diff
@@ -35,30 +35,30 @@ class RiskControllerTest(
         .jsonPath("$.personalDetailsOverview.age").isEqualTo("39")
         .jsonPath("$.personalDetailsOverview.gender").isEqualTo("Male")
         .jsonPath("$.personalDetailsOverview.crn").isEqualTo(crn)
-        .jsonPath("$.riskOfSeriousHarm.overallRisk").isEqualTo("HIGH")
-        .jsonPath("$.riskOfSeriousHarm.riskToChildren").isEqualTo("HIGH")
-        .jsonPath("$.riskOfSeriousHarm.riskToPublic").isEqualTo("HIGH")
-        .jsonPath("$.riskOfSeriousHarm.riskToKnownAdult").isEqualTo("HIGH")
-        .jsonPath("$.riskOfSeriousHarm.riskToStaff").isEqualTo("MEDIUM")
-        .jsonPath("$.riskOfSeriousHarm.lastUpdated").isEqualTo("2022-05-19")
+        .jsonPath("$.riskOfSeriousHarm.overallRisk").isEqualTo("")
```
Comment (jonnywyatt) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/controller/RiskControllerTest.kt:38]: Usually the MRD endpoints are sending blank values back as null

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/controller/RiskControllerTest.kt [hunk 3]
```diff
@@ -16,13 +16,13 @@ class RiskControllerTest(
 ) : IntegrationTestBase() {
 
   @Test
-  fun `retrieves risk data when no mappa details available`() {
+  fun `retrieves personal details data when no MAPPA or RoSH details available`() {
```
Comment (jonnywyatt) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/controller/RiskControllerTest.kt:19]: I can't see any assertion in this test that confirms the personal details are in the response?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RiskService.kt (hunk 1 above)
Comment (bills7281) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RiskService.kt:76]: Yes, that's what will happen 'fetchMappa(..)' here is inside a try/catch block which will swallow any 404 and allow the code execution to proceed with a 200.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/controller/RiskControllerTest.kt (hunk 3 above)
Comment (bills7281) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/controller/RiskControllerTest.kt:19]: image
They are there, see the image :)

**Review by jackmaloney — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RiskService.kt (hunk 1 above)
Comment (jonnywyatt) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RiskService.kt:76]: 👍

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/controller/RiskControllerTest.kt (hunk 3 above)
Comment (jonnywyatt) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/controller/RiskControllerTest.kt:19]: 👍

**Pull request merged**